qt/src/SourceManager.cc

```diff
@@ -44,6 +53,38 @@ QDataStream& operator>>(QDataStream& ds, Source*&) {
 	return ds;
 }
 
+Source::Source(const QString& _path, const QString& _displayname, bool _isTemp)
+				: path(_path), displayname(_displayname), isTemp(_isTemp) {
+	// Check is input PDF file encrypted
+	if(path.endsWith(".pdf")) {
+		Poppler::Document* document = Poppler::Document::load(path);
+		if(document) {
+			if (document->isLocked()) {
+				bool ok = false;
+				QString text;
+				while(true) {
+					text = QInputDialog::getText(MAIN, _("Enter password for the PDF"),
+														 _("Password:"), QLineEdit::Password,
+														 "", &ok);
+					document->unlock(text.toLocal8Bit(), text.toLocal8Bit());
+					if(!ok) {
+						throw std::invalid_argument("Encrypted PDF: skip file.");
+					}
+					if(!document->isLocked())
+					{
+						break;
+					} else {
+						QMessageBox::information(MAIN, _("Open encrypted PDF"), _("Entered password is wrong."));
+					}
+				}
+				if (ok && !text.isEmpty()){
+					password = text.toLocal8Bit();
```

Comment:
You can just assign password before the break; above and avoid this code block altogether

Reply:
fixed.